### ISSUE TITLE
[Question] New Feature support for old projects

### ISSUE THREAD

**Issue by paul1956** [dotnet/vblang#347]
In VB the compiler defaults to features from last major version, currently 15 and if you want to use features of 15.3 or 15.5 you have to specify the version specifically in the project file. When VB 16 come out anyone who didn't specify a specific version automatically gets VB 16 features, including features of 15.3 and 15.5. What happens if they had manually set the language version to 15.3 or 15.5 are those setting deleted, do VB 16 features not work, are they prompted to edit the files or something else?

**Comment by KathleenDollard** [comment]
If you have feedback on how this should work, let me know and I'll let the folks working on it know. 

We hope to have direct changes to VB version in the Visual Studio IDE (yeah!) 

We won't change explicit language versions automatically. You would have to do that via the project file or IDE. That would be true if you had explicitly set 15.0 as well - although that would be uncommon. 

I hope we can provide an upgrade path similar to C# - if you use a new feature you get a Code Fix to upgrade. That won't be clear until we know more about VB 16 features, because not all features are amenable to that approach.

**Comment by paul1956** [comment]
The only reason to set version explicitly is to get new features in minor release, I would hope if someone set 15.3 or 15.5 they would be offered option to remove them or at least a warning when 16.0 comes out during first time project opening. I can't imagine how turning on VB 16 would break a program that requested 15.3 or 15.5.
If someone doesn't specify they get VB16
If they specified 15.3 or 15.5 they get VB16 or at least a notification
 If they specified something less that 15.3 do nothing

**Comment by KathleenDollard** [comment]
Thanks for the thoughts. 

You point out that VB programmers selecting 15.3 or 15.5 probably intended to be selecting "Latest minor"

**Comment by zspitz** [comment]
@paul1956  From [Select the C# language version](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version):

> The C# compiler defaults to the latest major version of the language that has been released.

Same behavior as VB.NET.

**Comment by zspitz** [comment]
@paul1956 You mean like [this](https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/configure-language-version)?

**Comment by paul1956** [comment]
Exactly, thanks. It would be nice if the VS error had a linked to it. I, knew about lack of UI.

**Comment by paul1956** [comment]
Would a PR be accepted to change the text of the error to include the link?

**Comment by KathleenDollard** [comment]
@paul1956 The text of which error?

**Comment by paul1956** [comment]
The error returned for any feature that is not available because the Language Version of the Project is less than the current compiler and the user tries to us it and therefore requires use of the instructions.

CheckFeatureAvailability handles this error. It currently tells the user what version then need to use but not how to set it. One example of many.

![image](https://user-images.githubusercontent.com/12789369/48031651-c623f280-e109-11e8-9541-699fbca99fd0.png)
